lib/storage/data/external/AwsClient.js

```diff
@@ -26,47 +45,82 @@ class AwsClient {
         this._dataStoreName = config.dataStoreName;
         this._serverSideEncryption = config.serverSideEncryption;
         this._supportsVersioning = config.supportsVersioning;
-        this._vault = config.vault;
-        this._client = new AWS.S3(this._s3Params);
+        this._vault = config.vault;        
+        this._client = new S3Client(this._s3Params);
         this._logger = new werelogs.Logger('AwsClient');
     }
 
     setup(cb) {
-        // NOTE: to get the bucket location we interrogate the default
-        // region (us-east-1) and not the bucket's region (ref: ARTESCA-2141)
-        this._client.config.update({ region: 'us-east-1' });
-        // this request implicitly updates the endpoint for the location
-        // the following code explcitly sets it to avoid surprises
-        this._client.getBucketLocation({ Bucket: this._awsBucketName },
-            (err, res) => {
-                if (err && err.code !== 'AuthorizationHeaderMalformed') {
-                    this._logger.error('error during setup', {
-                        error: err,
-                        method: 'AwsClient.setup',
-                    });
-                    return cb(err);
-                }
-                let region;
-                if (err && err.code === 'AuthorizationHeaderMalformed') {
-                // set regional endpoint
-                    region = err.region;
-                } else if (res) {
-                    region = res.LocationConstraint;
-                }
-                this._client.config.update({ region });
+        const callback = typeof cb === 'function' ? cb : () => {};
+        let settled = false;
+        const done = err => {
+            if (settled) {
+                return;
+            }
+            settled = true;
+            callback(err);
+        };
```

Comment:
* this is actually a reimplementation of `jsutils.once()` : don't do this, just reuse
* why do we need this? Seems like a workaround of an actual issue...